x/programs/cmd/simulator/src/lib.rs

```diff
@@ -296,9 +331,20 @@ where
         self.writer.write_all(b"'\n")?;
         self.writer.flush()?;
 
-        let resp = self.responses.next().ok_or(ClientError::Eof)??;
+        self.responses.next().ok_or(ClientError::Eof)?
+    }
 
-        Ok(resp)
+    pub fn run_step<T>(
+        &mut self,
+        caller_key: &str,
+        step: &Step,
+    ) -> Result<PlanResponseTyped<T>, ClientError>
+    where
+        T: BorshDeserialize,
+    {
+        self._run_step(caller_key, step)?
+            .try_into()
+            .map_err(|_| ClientError::BorshDeserialization)
```

Comment:
We can do it in a follow-up PR, but we should really just let the end-user do the batching since it's all synchronous anyway. 

I think it's a bad practice to do a bunch of stuff and then do your assertions on the result. I suspect that more people (at least the ones following idiomatic testing practices) will be asserting on the results at each step.

Reply:
Absolutely, the plan is not adapted to our use-case anymore, this will be done when closing https://github.com/ava-labs/hypersdk/issues/981